src/transformers/models/marian/modeling_tf_marian.py

```diff
@@ -111,27 +111,34 @@ def _expand_mask(mask: tf.Tensor, tgt_len: Optional[int] = None, past_key_values
     return (1.0 - expanded_mask) * LARGE_NEGATIVE
 
 
-class TFMarianSinusoidalPositionalEmbedding(tf.keras.layers.Embedding):
+class TFMarianSinusoidalPositionalEmbedding(tf.keras.layers.Layer):
```

Comment:
This was causing trouble to XLA?

Reply:
Yes, more precisely, XLA on GPU. This is because the weights with the `tf.keras.layers.Embeddings` are initialized on CPU and the model is run on GPU, in XLA you cannot access from one to the other.

This is because the embeddings are created in the `__init__` of the classes instead of being created in the `build()`.